src/actions/clone.js

```diff
@@ -103,7 +103,7 @@ async function cloneFile(lock, ctx, params) {
   // add mark to the original file
   pipeline.hset(uploadKey, FILES_HAS_CLONES_FIELD, 1);
 
-  const provider = ctx.provider('remove', params);
+  const provider = ctx.provider('remove', uploadData);
```

Comment:
possible backward incompatibility

Reply:
maybe do something like

`ctx.provider('remove', params, uploadData)` and update ctx.provider signature to accept `userData` as third argument to the function

Reply:
so that b/c is preserved, but updated functions can account for new arguments